### DISCUSSION TITLE
Passing Uint16Array as image data in deck.gl v9

### DISCUSSION THREAD

**Discussion by ewandank** [visgl/deck.gl#8817] [Q&A]
In my app written with deck.gl v8, I had extended `bitmapLayer` to crush a 16-bit image to 8-bit to display within the fragment shader. 
To get a 16-bit image to render in deck.gl v8, I had to provide certain image props that are deprecated/no longer supported in v9. These props are as follows.

```js
const layer = new customBitmapLayer({
image: {
  data: props.data, //Uint16Array
  width: 512,
  height: 512,
  format: GL.RGBA16UI,
  dataFormat: GL.RGBA_INTEGER,
  type: GL.UNSIGNED_SHORT,
},
})
```
I am now trying to upgrade this to deck.gl v9. 
in v9, my understanding is that `type` and `dataFormat` are both deprecated.  I have tried  passing `rgba16uint`, `rgba16float`, and `rgba16unorm-webgl` as the `format` prop on the image, all of which result in no data making it to the `uniform sampler2D bitmapTexture` inside of the fragment shader.
 When passing a `Uint8Array` as the `image.data` prop with a format of `rgba8unorm` the custom shaders work as expected. 
i.e. 

```js
// This works
const layer = new customBitmapLayer({
image: {
  data: props.data, //Uint8Array
  width: 512,
  height: 512,
  format: "rgba8unorm",
},
})

// This doesn't work, rendering a black box where the image should be. 
const layer = new customBitmapLayer({
image: {
  data: props.data, //Uint16Array
  width: 512,
  height: 512,
  format: "rgba16unorm-webgl",
},
})
```

Any help would be much appreciated.

**Comment by ibgreen** [comment]
As a first guess, I'd suspect that there could be an issue in luma.gl with the mapping of that texture format to the three required WebGL parameters (which would be the parameters you used in your original example).

This would be somewhere in `webgl-texture.ts` ( the `WEBGLTexture` class), probably not too hard to debug. Since the `WEBGLTexture` class has been significantly refactored for 9.1 so we would probably need to make separate fixes on `master` and `9.0-release`. It may require more work to fix on 9.0.

Code
- [9.1 dev (`master`)](https://github.com/visgl/luma.gl/blob/master/modules/webgl/src/adapter/resources/webgl-texture.ts#L156)
- [`9.0-release`](https://github.com/visgl/luma.gl/blob/9.0-release/modules/webgl/src/adapter/resources/webgl-texture.ts#L280)

Unfortunately, luma.gl doesn't yet have a lot of test cases for 16-bit formats, something we'd love to see addressed.
